soc/xtensa/intel_adsp/ace/sram.c

```diff
@@ -28,9 +28,7 @@ __imr void hp_sram_init(uint32_t memory_size)
 		*(l2hsbpmptr + idx * 2) = 0;
 	}
 	for (idx = 0; idx < hpsram_ebb_quantity; ++idx) {
-		while (*(status + idx * 8) != 0) {
-			z_idelay(DELAY_COUNT);
-		}
+		while (*(status + idx * 8) != 0) {}
```

Comment:
This can silently spin forever if it fails, can we have a timeout and error code ?

Reply:
@lgirdwood it is a very basic init and at a very early stage (powering memory banks), can't imagine what can fail here (except extreme problems like HW failure)